src/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProvider.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\NelmioApiDoc\Extractor\AnnotationsProvider;
+
+use ApiPlatform\Core\Api\FilterCollection;
+use ApiPlatform\Core\Api\OperationMethodResolverInterface;
+use ApiPlatform\Core\Bridge\NelmioApiDoc\ApiPlatformParser;
+use ApiPlatform\Core\Hydra\ApiDocumentationBuilderInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Nelmio\ApiDocBundle\Extractor\AnnotationsProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Creates Nelmio ApiDoc annotations for the api platform.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ApiPlatformProvider implements AnnotationsProviderInterface
+{
+    private $resourceNameCollectionFactory;
+    private $apiDocumentationBuilder;
+    private $resourceMetadataFactory;
+    private $filters;
+    private $operationMethodResolver;
+    private $router;
+
+    public function __construct(
+        ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory,
+        ApiDocumentationBuilderInterface $apiDocumentationBuilder,
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        FilterCollection $filters,
+        OperationMethodResolverInterface $operationMethodResolver,
+        RouterInterface $router
+    ) {
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->apiDocumentationBuilder = $apiDocumentationBuilder;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->filters = $filters;
+        $this->operationMethodResolver = $operationMethodResolver;
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAnnotations() : array
```

Comment:
From the [RFC](https://wiki.php.net/rfc/return_types):
"If the parent does not declare a return type then the child is allowed to declare one."